### DISCUSSION TITLE
How to use library Modelica.Thermal.HeatTransfer.Components.BodyRadiation?

### DISCUSSION THREAD

**Discussion by bb6698** [OpenModelica/OpenModelica#11104] [Q&A]
Hi! I use library Modelica.Thermal.HeatTransfer.Components.BodyRadiation for studio thermal radiation between 2 bodys. I have a many problems with this library. 

Attached is my code: 

```mo
model thermal_analysis
  
  import Modelica.Thermal.HeatTransfer.Components.BodyRadiation;
  
  // Heat ports
  Modelica.Thermal.HeatTransfer.Interfaces.HeatPort_a port_Body1;
  Modelica.Thermal.HeatTransfer.Interfaces.HeatPort_b port_Body2
  
  BodyRadiation radiation(Gr=0.2);
equation
    // Connection of heat ports
    connect(port_Body1, radiation.port_a);
    connect(port_Body2, radiation.port_b);
end thermal_analysis;
```

And I want to simulate, I have this errors: 

```
[1] 11:24:15 Symbolic Error
[Modelica.Thermal.HeatTransfer.Interfaces.HeatPort: 4:3-5:75]: [Found equation without time-dependent variables: puertoLuna.Q_flow = 0.0](omeditmessagesbrowser:///Modelica.Thermal.HeatTransfer.Interfaces.HeatPort?lineNumber=4)

[2] 11:24:15 Translation Error
Internal error IndexReduction.pantelidesIndexReduction failed! System is structurally singular and cannot be handled because the number of unassigned equations is larger than the number of states. Use -d=bltdump to get more information.

[3] 11:24:15 Translation Error
Internal error Transformation Module PFPlusExt index Reduction Method Pantelides failed!
```

I don't know how to fix it and make a model correctly to be able to use the library.I need to help!! 

Thanks

**Comment by casella** [comment]
This model has unconnected top-level thermal ports, for which default equations of the flow variables (i.e., Q_flow thermal powers) are set to zero. Hence, it is completely isolated. It is therefore obviously singular, because it is impossible to determine what are the temperatures. You should connect it to something that has a temperature, e.g. a fixed temperature source, or a body with some thermal capacitance.

**Reply by bb6698** [reply]
Hi! Thank for your answer.  I will try it!

If it wouldn't be too much trouble, could you give me a basic code example of how to use BodyRadiation? I would really appreciate it, as I am just starting to learn openmodelica and its different libraries :) 

Cheers

**Comment by casella** [comment]
Here it is: a model of an electric toaster [Example.zip](https://github.com/OpenModelica/OpenModelica/files/12551045/Example.zip).

It represents the heat source (a hot wire at 700 °C), radiative heat transfer with the bread slice, the bread slice heat capacity, and the convective heat transfer from bread slice to the ambient.

Of course it is highly simplified: there's no accounting for water evaporation and for the non-uniform temperature distribution between the surface and the core of the bread slice. But I hope it helps understanding how Modelica works.
